Tools/TestWebKitAPI/Tests/WebKitCocoa/ShouldOpenAppLinks.mm

```diff
@@ -0,0 +1,138 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+
+#import "HTTPServer.h"
+#import "Utilities.h"
+#import <WebKit/WKNavigationActionPrivate.h>
+#import <WebKit/WKWebViewPrivate.h>
+#import <WebKit/WebKit.h>
+
+@interface ShouldOpenAppLinksTestNavigationDelegate: NSObject<WKNavigationDelegate>
+@property (nonatomic, retain) WKNavigationAction *lastNavigationAction;
+- (void)doAndWaitForNavigationToFinish:(void (^NS_NOESCAPE)(void))block;
+@end
+
+@implementation ShouldOpenAppLinksTestNavigationDelegate {
+    bool _hasNavigationFinished;
+}
+
+- (void)doAndWaitForNavigationToFinish:(void (^NS_NOESCAPE)(void))block
+{
+    _hasNavigationFinished = false;
+    block();
+    TestWebKitAPI::Util::run(&_hasNavigationFinished);
+}
+
+- (void)dealloc
+{
+    [_lastNavigationAction release];
+    [super dealloc];
+}
+
+- (void)webViewWebContentProcessDidTerminate:(WKWebView *)webView
+{
+    [webView reload];
+}
+
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler
+{
+    [_lastNavigationAction release];
+    _lastNavigationAction = [navigationAction retain];
+
+    decisionHandler(WKNavigationActionPolicyAllow);
+}
+
+- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation
+{
+    _hasNavigationFinished = true;
+}
+
+- (void)webView:(WKWebView *)webView didFailProvisionalNavigation:(WKNavigation *)navigation withError:(NSError *)error
+{
+    FAIL();
+}
+
+- (void)webView:(WKWebView *)webView didFailNavigation:(WKNavigation *)navigation withError:(NSError *)error
+{
+    FAIL();
+}
+
+@end
+
+static TestWebKitAPI::HTTPServer testServer(void)
```

Comment:
(void) should be ()
Let's give this a slightly more specific name, like appLinkTestServer.  With unified builds this name is likely to collide.